fix: deb/apk packager set owner & group failed; feat: add EmptyFolder struct.

Serinalice opened the pull request:
- Bug: During testing, I found that the owner and group settings of content do not take effect on apk and debian.
- Fix: `FileInfo`'s `Sys()` funciton returns `nil`, when use `tar.FileInfoHeader(file, file.Source)` the `Uname` and `Gname` attributes of the header will not be set, so I set them manually.
- Feature: Users should be able to manually control the metadata of the empty folders. For this reason, I added the `EmptyFolder` struct which replaces `string`.

caarlos0 commented:
hey, thanks for the PR!

Will review it shortly :)

> * Feature: Users should be able to manually control the metadata of the empty folders. For this reason, I added the `EmptyFolder` struct which replaces `string`.

meanwhile, can we split this into another PR + add related docs?

caarlos0 commented on files/files.go:
```diff
@@ -114,6 +114,36 @@ func (c *Content) Sys() interface{} {
+
+type EmptyFolders []*EmptyFolder
+
+func (f *EmptyFolder) WithFolderInfoDefaults() *EmptyFolder {
```
this should probably happen on the "global" `WithDefaults` function... iterate the empty folders list and set defaults as needed...

Serinalice replied:
Oh,Sorry...I missed the function `WithDefaults `. I will reconsider that and update.

caarlos0 commented on nfpm.go:
```diff
@@ -248,18 +248,18 @@ func (i *Info) parseSemver() {
+	Suggests     []string           `yaml:"suggests,omitempty" jsonschema:"title=suggests directive,example=nfpm"`
+	Conflicts    []string           `yaml:"conflicts,omitempty" jsonschema:"title=conflicts directive,example=nfpm"`
+	Contents     files.Contents     `yaml:"contents,omitempty" jsonschema:"title=files to add to the package"`
+	EmptyFolders files.EmptyFolders `yaml:"empty_folders,omitempty" jsonschema:"title=empty folders to be created when installing the package,example=/var/log/nfpm"`
```
example in jsonschema is no longer valid as this is now a struct.

Serinalice replied:
Yeah,I will fix this.

caarlos0 commented on files/files.go:
```diff
@@ -114,6 +114,36 @@ func (c *Content) Sys() interface{} {
 }
 
+type EmptyFolder struct {
+	Path  string      `yaml:"path,omitempty"`
+	Owner string      `yaml:"owner,omitempty"`
+	Group string      `yaml:"group"`
+	Mode  os.FileMode `yaml:"mode,omitempty"`
+	MTime time.Time   `yaml:"mtime,omitempty"`
```
missing the jsonschema tags

Serinalice replied:
I think maybe it is a good idea to keep the `EmptyFolder` consistent with `Content`/`ContentInfo`. 
Are jsonschema tags necessary?

djgilcrease replied:
I would try to make this Just part of the content structure with an `emptyfolder` type. And leave the current empty folder structure alone for backwards compatibility. We can then depreciate the empty folder structure.

caarlos0 replied:
we could also write a custom decoder for this type, so the old format (string only) still works...

Serinalice commented:
> hey, thanks for the PR!
> 
> Will review it shortly :)
> 
> > * Feature: Users should be able to manually control the metadata of the empty folders. For this reason, I added the `EmptyFolder` struct which replaces `string`.
> 
> meanwhile, can we split this into another PR + add related docs?

Yes, no problem.
